frontend/src/pages/AllChatsPage.jsx

```diff
@@ -23,16 +23,50 @@ const ChatInboxHeader = () => (
     </div>
 )
 
+// find the blocked accounts of the user
+function useFetchBlockedUsers(myUid) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+    // create a blocked user dictionary
+    const reduceList = blockedUsers.reduce((accumulator, item) => {
+        if (item.blocked === true) {
+            accumulator[item.uid] = item.blocked
+        }
+        return accumulator
+    }, {})
+    return reduceList
+}
+
 export default function AllChatsPage() {
     const authUser = useCurrentAuthUser()
     const matches = useGetAllUserMatches(authUser?.uid)
+    const blockedUsers = useFetchBlockedUsers(authUser?.uid)
+
+    const unblockedMatches = matches.filter(
+        (match) =>
+            !Object.keys(match.participants).some((participant) =>
+                Object.keys(blockedUsers).includes(participant)
```

Comment:
Here, you can use `participant in blockedUsers` as the condition, which should check whether or not the key exists in constant time, rather than linear time to get all the keys, and then search if the key is found in the list.